security/pkg/server/grpc/server.go

```diff
@@ -142,7 +142,7 @@ func New(ca ca.CertificateAuthority, ttl time.Duration, hostname string, port in
 
 func (s *Server) createTLSServerOption() grpc.ServerOption {
 	cp := x509.NewCertPool()
-	cp.AppendCertsFromPEM(s.ca.GetRootCertificate())
```

Comment:
Do we need to append both root certificate and certificate chain?
If we only need to append certificate chain, then --cert-chain argument should be mandatory.

Reply:
root cert is always a single cert, everything else should go to workload side, i.e., 

cert-chain.pem should be a chain 
ca-root.pem should be a single cert.

Reply:
Reverted unnecessary changes